RentACar/RentACar/Intefaces/IRepository.cs

```diff
@@ -0,0 +1,13 @@
+﻿using System.Collections.Generic;
+
+namespace RentACar.Intefaces
+{
+    public interface IRepository<T>
```

Comment:
I missed this earlier: I think you need to add the generic constraints here as well (not sure about his but please double check and let me know)